lib/index.d.ts

```diff
@@ -1,98 +1,507 @@
+/* This file was automatically generated by json-schema-to-typescript. DO NOT MODIFY IT BY HAND. */
+
+/**
+ * the entities that the subscription provides notifications for
+ */
+export type Include = string[]
+/**
+ * provider actions for this specific add-on
+ */
+export type Actions = {
+  [k: string]: unknown
+  /**
+   * identifier of the action to take that is sent via SSO
+   */
+  action?: string
+  /**
+   * a unique identifier
+   */
+  id?: string
+  /**
+   * the display text shown in Dashboard
+   */
+  label?: string
+  /**
+   * if the action requires the user to own the app
+   */
+  requires_owner?: boolean
+  /**
+   * absolute URL to use instead of an action
+   */
+  url?: string
+} & {
+  [k: string]: unknown
+}[]
+/**
+ * config vars exposed to the owning app by this add-on
+ */
+export type ConfigVars = string[]
+/**
+ * the compliance regimes applied to an add-on plan
+ */
+export type Compliance = ('HIPAA' | 'PCI')[] | null
+/**
+ * errors associated with invalid app.json manifest file
+ */
+export type ManifestErrors = string[]
+/**
+ * result of postdeploy script
+ */
+export type Postdeploy = {
+  [k: string]: unknown
```

Comment:
I have strong reservations about this. The ability to append arbitrary keys and make all properties optional to every interface within this schema is generally not useful.

This is a Postdeploy object according to this schema. It's also a PlatformAddOnPlanAction. It's also literally any type in this schema.
    ```
const actions: Postdeploy = { notThere: 'myValue' };
```
This usually indicates a mistake but TS allows it to go on it's merry way.

Also, this will mandate the use of the optional chaining operator when reading keys:
```
const postdeploy: Postdeploy = {exit_code: 0};
if (postdeploy?.exit_code === 0) {
  // Do stuff
}
```
PostDeploy.exit_code is also listed as `readonly` in the schema but it's optional/read/write in the interface
```
"exit_code": {
              "description": "The exit code of the postdeploy script",
              "example": 1,
              "readOnly": true,
              "type": [
                "integer"
              ]
            }
```